### PR TITLE
Update `SalaryDetails` to query currencies and pay types

### PR DESCRIPTION
**Pull request by AkiraJ48**
This is in reference to https://github.com/SEEK-Jobs/indie-ryanair-frontend/pull/883#pullrequestreview-1513491646 and implements the queries in https://github.com/SEEK-Jobs/indie-graphql-server/pull/2249.

### CODE REVIEW

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx [hunk 1]
```diff
@@ -87,6 +122,23 @@ export const SalaryDetails = ({
     min: validateMinAmount(min, errors),
   };
 
+  if (
+    payTypesLoading ||
+    currenciesLoading ||
+    payTypesData === undefined ||
+    currenciesData === undefined
+  ) {
+    return <Loader />;
```
Comment (72636c) [inline fe/lib/components/SalaryDetails/SalaryDetails.tsx:131]: Nit: maybe not for this PR, but do we want a nicer loading state that results in a smaller layout shift when the data is finally loaded?

I think "Pay details" as a heading for example would be useful to keep around even while loading.

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx [hunk 2]
```diff
@@ -87,6 +122,23 @@ export const SalaryDetails = ({
     min: validateMinAmount(min, errors),
   };
 
+  if (
+    payTypesLoading ||
+    currenciesLoading ||
+    payTypesData === undefined ||
+    currenciesData === undefined
```
Comment (72636c) [inline fe/lib/components/SalaryDetails/SalaryDetails.tsx:129]: Do these need to explicitly test for `undefined`, or can we treat `null`s the same way?

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx [hunk 3]
```diff
@@ -87,6 +122,23 @@ export const SalaryDetails = ({
...
+    return <Loader />;
+  }
+
+  const payTypes = payTypesData.payTypes.reduce<Record<string, PayType>>(
+    (acc, data) => ({
+      ...acc,
+      [`${data.basisCode}.${data.intervalCode}`]: data,
+    }),
+    {},
+  );
```
Comment (72636c) [inline fe/lib/components/SalaryDetails/SalaryDetails.tsx:140]: Is this pass necessary? I would expect the query response to be deduplicated and in a reasonable order.

**Review by 72636c — APPROVED** [review]
🙇

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx (hunk 3 above)
Comment (AkiraJ48) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx:140]: Do you mean to just use the original response? I mainly adjust the shape so that it was easier for https://github.com/seek-oss/wingman/pull/959/files/448d70dbd088ab8055c79f4e187aac0d3580b146#diff-de34c4e404c87e05543daa353d41e70903945fea108a2430232043cd03912cabR266

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx (hunk 2 above)
Comment (AkiraJ48) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx:129]: Realistically we should never get a `null` response with these 2 queries, if we did something went very wrong 😅 .

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx (hunk 3 above)
Comment (72636c) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx]: Oh I see, I was only looking at the `Object.entries` usage below. Would it make sense to change this to a `payType = data.payTypes.find((p) => p.label === payTypeLabel)` and simplify that part to `data.payTypes.map(...)`?

**Review by 72636c — APPROVED** [review]

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx (hunk 2 above)
Comment (72636c) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx]: Is the specific behaviour around `undefined` and not `null` guaranteed by the GraphQL server/client? If not I'd just err on the side of treating them the same in case someone decides to change the implementation in future.

File: fe/lib/components/SalaryDetails/SalaryDetails.tsx (hunk 3 above)
Comment (AkiraJ48) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx]: Did you mean something like https://github.com/seek-oss/wingman/pull/959/commits/564ebf85389d8d823aa392403f875353a246079b?

Comment (72636c) [reply fe/lib/components/SalaryDetails/SalaryDetails.tsx]: Ooh I see, there's a bit of complexity as we need the codes and the labels in different places. I think I'm fine with either approach having seen that, thanks for looking into this 🙇

**Review by 72636c — APPROVED** [review]
